sources/TileDB.CSharp/FragmentInfo.cs

```diff
@@ -0,0 +1,681 @@
+using System;
+using System.Runtime.CompilerServices;
+using TileDB.CSharp.Marshalling;
+using TileDB.CSharp.Marshalling.SafeHandles;
+using TileDB.Interop;
+
+namespace TileDB.CSharp
+{
+    /// <summary>
+    /// Represents a TileDB fragment info object.
+    /// </summary>
+    public unsafe sealed class FragmentInfo : IDisposable
+    {
+        private readonly Context _ctx;
+        private readonly FragmentInfoHandle _handle;
+
+        internal FragmentInfoHandle Handle => _handle;
+
+        /// <summary>
+        /// Creates a <see cref="FragmentInfo"/> object.
+        /// </summary>
+        /// <param name="ctx">The <see cref="Context"/> associated with this object.</param>
+        /// <param name="uri">The URI of the array to load the fragment info for.</param>
+        public FragmentInfo(Context ctx, string uri)
+        {
+            _ctx = ctx;
+            using var uri_ms = new MarshaledString(uri);
+            _handle = FragmentInfoHandle.Create(ctx, uri_ms);
+        }
+
+        /// <summary>
+        /// Disposes this <see cref="FragmentInfo"/> object.
+        /// </summary>
+        public void Dispose()
+        {
+            _handle.Dispose();
+        }
+
+        /// <summary>
+        /// Loads the fragment info.
+        /// </summary>
+        public void Load()
+        {
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            _ctx.handle_error(Methods.tiledb_fragment_info_load(ctxHandle, handle));
+        }
+
+        /// <summary>
+        /// Loads the fragment info from an encrypted array.
+        /// </summary>
+        /// <param name="encryptionType">The encryption type to use.</param>
+        /// <param name="key">The encryption key to use.</param>
+        public void LoadWithKey(EncryptionType encryptionType, ReadOnlySpan<byte> key)
+        {
+            using var ctxHandle = _ctx.Handle.Acquire();
+            using var handle = _handle.Acquire();
+            fixed (byte* keyPtr = key)
+            {
+                _ctx.handle_error(Methods.tiledb_fragment_info_load_with_key(ctxHandle, handle, (tiledb_encryption_type_t)encryptionType, keyPtr, (uint)key.Length));
```

Comment:
This API is obsolete. The keys are now supposed to be set through the `Config`, right?

Reply:
Removed it after seeing the deprecation policy.